pkg/scaffold/cmd.go

```diff
@@ -129,4 +135,18 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func setupMetrics(mgr manager.Manager) {
```

Comment:
We may have already discussed this, but should we put this functionality in the metrics package so that other code (e.g. Ansible and Helm operators) can use it?

Given the name, does it make sense to have `metrics.ExposeMetricsPort` call `client.Create`?

Reply:
> We may have already discussed this, but should we put this functionality in the metrics package so that other code (e.g. Ansible and Helm operators) can use it?

No, we haven't but open to suggestions!

Reply:
Since we're generating this wrapper and expecting it be used as is, we might as well move `client.Create()` into `metrics.ExposeMetricsPort()` and just use that. 

We can have `ExposeMetricsPort()` return an error and the created service object (or the pre-existing service object via `client.Get()`). And in `cmd/manager/main.go` we can log the returned error.

Reply:
Sure,my thought behind it was to let the user decide when to create the `Service`. I think in this case we should maybe at least leave a comment that this will create a Service object and expose the metrics port, wdyt?

Reply:
I'd probably lean towards putting `client.Create()` in `metrics.ExposeMetricsPort()`.

If later it turns out that there would be value in splitting out the Service object creation
and the creation of it in the cluster, we can add a function to the `metrics` package that does just the Service object creation (and then we could refactor `metrics.ExposeMetricsPort()` to use it).

(Sorry if that's confusing. The word "creation" is a bit overloaded in this context :smile:)

Reply:
@joelanford 
> I'd probably lean towards putting client.Create() in metrics.ExposeMetricsPort().

I already did :)